homeassistant/components/awair/sensor.py

```diff
@@ -1,248 +1,250 @@
+        users don't have to handle positive/negative values that ultimately "mean"
+        the same thing.
 
-class AwairData:
-    """Get data from Awair API."""
+        https://docs.developer.getawair.com/?version=latest#awair-score-and-index
+        """
+        attrs = {}
+        label = slugify(SENSOR_TYPES[self.__kind][ATTR_DEVICE_CLASS])
 
-    def __init__(self, client, uuid, throttle):
-        """Initialize the data object."""
-        self._client = client
-        self._uuid = uuid
-        self.data = {}
-        self.attrs = {}
-        self.async_update = Throttle(throttle)(self._async_update)
+        if self.__kind in self.__air_data.indices:
+            attrs[f"{label}_awair_index"] = abs(self.__air_data.indices[self.__kind])
+        elif self.__kind in DUST_ALIASES and API_DUST in self.__air_data.indices:
+            attrs[f"{label}_awair_index"] = abs(self.__air_data.indices.dust)
 
-    async def _async_update(self):
-        """Get the data from Awair API."""
-        resp = await self._client.air_data_latest(self._uuid)
+        return attrs
 
-        if not resp:
-            return
+    @property
+    def device_info(self) -> dict:
+        """Device information."""
+        info = {
+            "identifiers": {(DOMAIN, self.__device.uuid)},
+            "manufacturer": "Awair",
+            "model": self.__device.model,
+        }
 
-        timestamp = dt.parse_datetime(resp[0][ATTR_TIMESTAMP])
-        self.attrs[ATTR_LAST_API_UPDATE] = timestamp
-        self.data[ATTR_SCORE] = resp[0][ATTR_SCORE]
+        if self.__device.name:
+            info["name"] = self.__device.name
 
-        # The air_data_latest call only returns one item, so this should
-        # be safe to only process one entry.
-        for sensor in resp[0][ATTR_SENSORS]:
-            self.data[sensor[ATTR_COMPONENT]] = round(sensor[ATTR_VALUE], 1)
+        if self.__device.mac_address:
+            info["connections"] = {
+                (dr.CONNECTION_NETWORK_MAC, self.__device.mac_address)
+            }
 
-        _LOGGER.debug("Got Awair Data for %s: %s", self._uuid, self.data)
+        return info
+
+    async def async_added_to_hass(self) -> None:
+        """Connect to dispatcher listening for entity data notifications."""
+        self.__coordinator.async_add_listener(self.async_write_ha_state)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Disconnect from update signal."""
+        self.__coordinator.async_remove_listener(self.async_write_ha_state)
+
+    async def async_update(self) -> None:
+        """Update Awair entity."""
+        await self.__coordinator.async_request_refresh()
+
+    @property
+    def __air_data(self) -> Optional[AwairResult]:
+        """Return the latest data for our device, or None."""
+        result: Optional[AwairResult] = self.__coordinator.data.get(
+            self.__device.uuid, None
+        )
+        if result:
+            return result.air_data
```

Comment:
```suggestion
        if result:
            return result.air_data
        else:
            return None
```

Reply:
That's fine by me, if that's your stylistic preference. Given that a python method otherwise returns None at the end of a method, it doesn't actually change the logic.